### PR TITLE
[core] remove PythonGcsClient as well as Python gRPC stubs to GCS.

### PR DESCRIPTION
**Pull request by rynewang**
After the effort to add Async GcsClient binding to Python, we allow both old and new code paths to live for a while and now it seems to be pretty stable. We never received any issues from customers about the NewGcsClient. Hence we remove the old one.

### CODE REVIEW

File: python/ray/_private/gcs_aio_client.py [hunk 1]
```diff
@@ -1,26 +1,7 @@
...
-# can happen. Extra requests will need to wait for the existing requests to finish.
-# Executor rules:
-# If the arg `executor` in GcsAioClient constructor is set, use it.
-# Otherwise if env var `GCS_AIO_CLIENT_DEFAULT_THREAD_COUNT` is set, use it.
-# Otherwise, use 5.
-# This is only used for the OldGcsAioClient.
-GCS_AIO_CLIENT_DEFAULT_THREAD_COUNT = env_integer(
-    "GCS_AIO_CLIENT_DEFAULT_THREAD_COUNT", 5
-)
+from ray._raylet import NewGcsClient
```
Comment (jjyao) [inline python/ray/_private/gcs_aio_client.py:4]: NewGcsClient -> GcsClient?

Comment (rynewang) [reply python/ray/_private/gcs_aio_client.py:4]: Yeah problem is we have `NewGcsClient` the binding and the `GcsClient` the one with call frequency counter. Not sure about the naming here...

File: python/ray/_private/gcs_aio_client.py [hunk 2]
```diff
@@ -29,24 +10,11 @@ class GcsAioClient:
...
 
-    This class is in transition to use the new C++ GcsClient binding. The old
-    PythonGcsClient binding is not deleted until we are confident that the new
-    binding is stable.
-
-    Defaults to the new binding. If you want to use the old binding, please
-    set the environment variable `RAY_USE_OLD_GCS_CLIENT=1`.
+    Historical note: there was a `ray::gcs::PythonGcsClient` C++ binding which has only
+    sync API and in Python we wrap it with ThreadPoolExecutor. It's been removed in
+    favor of `ray::gcs::GcsClient` which contains async API.
```
Comment (jjyao) [inline python/ray/_private/gcs_aio_client.py:15]: Do we need this lol

File: python/ray/_private/gcs_aio_client.py (hunk 1 above)
Comment (jjyao) [reply python/ray/_private/gcs_aio_client.py:4]: Got it. 

Should it be `GcsClient`, `GcsSyncClient` and `GcsAsyncClient`?

**Review by jjyao — APPROVED** [review]

File: python/ray/_private/gcs_aio_client.py (hunk 1 above)
Comment (rynewang) [reply python/ray/_private/gcs_aio_client.py:4]: There are ~150 `GcsClient` needs to be renamed to GcsSyncClient, which is too big a change. Instead, I propose:

- GcsClient: sync
- GcsAioClient: async
- InnerGcsClient: binding for both

and it's apparent users should not use InnerGcsClient.

File: python/ray/_private/gcs_aio_client.py (hunk 2 above)
Comment (rynewang) [reply python/ray/_private/gcs_aio_client.py:15]: removed

**Pull request merged**